src/ml/dimensionality_reduction/pca.nim

```diff
@@ -25,4 +26,19 @@ proc pca*[T: SomeReal](x: Tensor[T], nb_components = 2): Tensor[T] =
   let (_, eigvecs) = cov_matrix.symeig(true, ^nb_components .. ^1) # Note: eigvals/vecs are returned in ascending order
 
   # [Nb_obs, Nb_feats] * [Nb_feats, Nb_components], don't forget to reorder component in descending order
-  result = mean_centered * eigvecs[_, ^1..0|-1]
+  var
+    descending_eigvecs = eigvecs[_, ^1..0|-1]
+    pca_results = mean_centered * descending_eigvecs
+
+  return (pca_results, descending_eigvecs)
```

Comment:
Regarding lines: https://github.com/metasyn/Arraymancer/blob/98a2b4e4812bb3b0f2e203005871db255f216ca1/src/ml/dimensionality_reduction/pca.nim#L29-L33

Thanks for the change.

For performance this should be written:

```Nim
result.components = eigvecs[_, ^1..0|-1]
result.results = mean_centered * result.components
```

In your case you allocate once for descending_eigvecs, once for pca_result
And
  - it might be that you allocated once for (result_tuple), if GCC didn't put them adjacent in memory.
     it's only Tensor metadata but still.
  - fortunately there won't be 2 allocations for both tensors data because tensors have ref semantics.